src/main/java/org/cryptomator/common/PropertiesPreprocessor.java

```diff
@@ -0,0 +1,66 @@
+package org.cryptomator.common;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public class PropertiesPreprocessor {
+
+	private static final Logger LOG = LoggerFactory.getLogger(PropertiesPreprocessor.class);
+	private static final Pattern TEMPLATE = Pattern.compile("@\\((.+)\\)");
+	private static final LoggingEnvironment ENV = new LoggingEnvironment(System.getenv(), LOG);
+
+	public static void run() {
+		var properties = System.getProperties();
+		properties.stringPropertyNames().stream() //
+				.filter(s -> s.startsWith("cryptomator.")) //
+				.forEach(key -> {
+					var value = properties.getProperty(key);
+					var newValue = process(value);
+					if(! value.equals(newValue)) {
+						LOG.info("Changed property {} from {} to {}.", key, value, newValue);
+					}
+					properties.setProperty(key, newValue);
+				});
+		LOG.info("Preprocessed cryptomator properties.");
+	}
```

Comment:
Please don't actually "change" the properties. Rather run the `process` step when reading from properties inside `Environment.java`

Reply:
As already said over a different channel: This won't be sufficient for underlying libraries, since they cannot access the Environment class of Cryptomator. An idea would be to add another API to integrations-api to read such properties.

My argument is, that we are only changing properties of the Cryptomator namespace, we do it at the main entry point before loading any additional classes and the Properties API even allows such changes. Additionally, the properties are fixed at application start. If we evaluate them dynamically, we have less guarantees.

Reply:
For the sake of agility, we will use the current seek and replace approach. I'll open a seperate issue to put this coupling between Cryptomator and its libraries into a more formal (api) way.